scripts/elasticsearch_nodes.py

```diff
@@ -0,0 +1,130 @@
+map of string values, so the list of node IDs is JSON encoded into a string and
+later converted back into a list when processed in the Terraform config.
+
+Example output when using a template that has four node ID placeholders and a
+current deployment that has two ES nodes:
+{"nodes": "[\"tVsGH2BSXXYe9sMLEYRcIv\", \"ydG9meZMAGkTnAlPgLljz8\", \"blank\", \"blank\"]"}
+"""
+from itertools import (
+    repeat,
+)
+import json
+import logging
+from time import (
+    sleep,
+)
+from typing import (
+    Sequence,
+)
+
+from botocore.exceptions import (
+    ClientError,
+)
+import jq
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+log = logging.getLogger(__name__)
+
+# How many times to try getting the node IDs from AWS before giving up.
+max_attempts = 10
+
+
+def main() -> None:
+    node_ids = get_node_ids()
+    data = prepare_node_ids(node_ids)
+    print(json.dumps(data))
+
+
+def get_node_ids() -> Sequence[str]:
+    """
+    Return a list of ES node IDs used by the current deployment.
+    """
+    for i in range(max_attempts):
+        log.debug(f'Attempt {i}/{max_attempts}')
+        try:
+            response = aws.cloudwatch.list_metrics(Namespace='AWS/ES',
+                                                   MetricName='CPUUtilization',
+                                                   RecentlyActive='PT3H',
+                                                   Dimensions=[
+                                                       {
+                                                           'Name': 'DomainName',
+                                                           'Value': config.es_domain
+                                                       }
+                                                   ])
+        except ClientError as e:
+            log.warning(e.response['Error']['Code'] + '. Retrying...')
+        else:
+            nodes = [
+                d['Value']
+                for m in response['Metrics']
+                for d in m['Dimensions']
+                if d['Name'] == 'NodeId'
+            ]
+            if len(nodes) > 0:
+                return nodes
+            else:
+                log.warning('No node IDs found in response.')
+        sleep(5)
+
+    raise RuntimeError('Failed to obtain node IDs. Max attempts exhausted.')
+
+
+def prepare_node_ids(node_ids: Sequence[str]) -> dict[str, str]:
+    """
+    Return the given node IDs in a data structure that TerraForm can use as an
+    external data source.
+    """
+    # The list of nodes IDs is appended with dummy entries to ensure the length
+    # matches the number of placeholders in the template. If our list had less
+    # than the template, Terraform would throw an "Invalid index" error. Also,
+    # the dummy entries cannot be empty strings.
+    placeholder_count = node_placeholder_count()
```

Comment:
```suggestion
    num_placeholders = node_placeholder_count()
```

You've been around long enough to know that this is what we prefer.